### PR TITLE
Setup workflows and Elixir package

### PR DESCRIPTION
**Pull request by jonatanklosko**
### Builds

Upon a new release a workflow builds XLA for several targets (all x86_64):

* linux cpu, tpu, cuda (10.2, 11.0, 11.1+)
* macos cpu
* windows cpu

After a successful build we would publish the new version to hex, so then it's actually available to exla.

### Package

`XLA.archive_path!()` returns a path to `.tar.gz` archive, that's how exla would obtain the archive. Depending on the environment variables, this call either downloads a matching archive, or builds locally, or downloads for a user-specified url. In all cases the archives land in `cache/` and are reused for all environments.

---

The windows build was quite challenging, but finally I got something that passes, yet to figure out how to integrate it with exla compilation though.

### CODE REVIEW

File: lib/xla.ex [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+      true ->
+        path = archive_path_for_matching_download()
+        unless File.exists?(path), do: download_matching!(path)
+        path
+    end
+  end
+
+  defp build?() do
+    System.get_env("XLA_BUILD") == "true"
```
Comment (josevalim) [inline lib/xla.ex:45]:
```suggestion
    System.get_env("XLA_BUILD") in ~w(1 true)
```

?

File: lib/mix/tasks/xla.release_tag.ex [hunk 2]
```diff
@@ -0,0 +1,12 @@
+defmodule Mix.Tasks.Xla.ReleaseTag do
```
Comment (josevalim) [inline lib/mix/tasks/xla.release_tag.ex:1]: What about a single task that prints both information? Or maybe a single task that prints both but allows a parameter to control to print only one of the them?

**Comment by josevalim** [comment]
Some comments added. This is awesome!!! 😍

File: lib/mix/tasks/xla.release_tag.ex (hunk 2 above)
Comment (jonatanklosko) [reply lib/mix/tasks/xla.release_tag.ex:1]: We need to get one at a time, but a parameter sounds good!!

Comment (josevalim) [reply lib/mix/tasks/xla.release_tag.ex]: If those are also public functions, then someone can also do `mix run -e "IO.puts XLA..."`. Who is using those tasks?

Comment (jonatanklosko) [reply lib/mix/tasks/xla.release_tag.ex]: We're using them in the scripts (mostly on the CI), like `tag=$(mix xla.info release_tag)`. Calling like that works too, but I prefer the compiler checks we get by making this into a task and explicitly specifying what information the scripts need :)

Comment (josevalim) [reply lib/mix/tasks/xla.release_tag.ex]: I see. Your call then! :)

**Review by josevalim — APPROVED** [review]

**Comment by seanmor5** [comment]
Fantastic! I can update the Windows EXLA build process so no need to worry about that

**Comment by jonatanklosko** [comment]
@seanmor5 that would be totally lovely <3 I submitted an integration in https://github.com/elixir-nx/nx/pull/453, so it's a matter of getting `mix compile` to work on windows :)

**Comment by jonatanklosko** [comment]
Just to establish the next steps, I think we should:
1. Patch https://github.com/elixir-nx/nx/pull/453 with windows
2. Ship this PR with updated version/repo references, publish v0.1.0, wait for the build, publish to hex
3. Update https://github.com/elixir-nx/nx/pull/453 to point to hex, so we can merge

If I miss anything let me know :D

**Comment by josevalim** [comment]
I think we can merge this and merge the Nx one too, so people can give feedback while we fix the windows one.

**Comment by josevalim** [comment]
Merge this and do a release. :)

**Comment by jonatanklosko** [comment]
Done, the build is running :)

**Pull request merged**